hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1066,17 +1097,49 @@ private boolean validateTimelineBeforeSchedulingCompaction(Option<String> inFlig
     // any instants before that is already synced with metadata table.
     // c. Do consider out of order commits. For eg, c4 from DT could complete before c3. and we can't trigger compaction in MDT with c4 as base instant time, until every
     // instant before c4 is synced with metadata table.
-    List<HoodieInstant> pendingInstants = dataMetaClient.reloadActiveTimeline().filterInflightsAndRequested()
-        .findInstantsBeforeOrEquals(latestDeltaCommitTimeInMetadataTable).getInstants();
 
-    if (!pendingInstants.isEmpty()) {
-      checkNumDeltaCommits(metadataMetaClient, dataWriteConfig.getMetadataConfig().getMaxNumDeltacommitsWhenPending());
+    // There should not be any incomplete instants on MDT
+    HoodieActiveTimeline metadataTimeline = metadataMetaClient.reloadActiveTimeline();
```

Comment:
we would have reloaded the timeline just before this call at the beginning of performTableServices. 
Do we still need another reload here? 
Lets be judicious about metaclient reloads.